lib/networking/vpc-stack.ts

```diff
@@ -0,0 +1,96 @@
+/* Copyright OpenSearch Contributors
+SPDX-License-Identifier: Apache-2.0
+
+The OpenSearch Contributors require contributions made to
+this file be licensed under the Apache-2.0 license or a
+compatible open source license. */
+
+import { Construct } from 'constructs';
+import {
+  IPeer,
+  ISecurityGroup,
+  IVpc,
+  Peer, Port, SecurityGroup, SubnetType, Vpc,
+} from 'aws-cdk-lib/aws-ec2';
+import { App, Stack, StackProps } from 'aws-cdk-lib';
+
+export interface vpcProps extends StackProps{
+    cidrBlock: string,
+    maxAzs: number,
+    vpcId: string,
+    securityGroupId: string,
+    serverAccessType: string,
+    restrictServerAccessTo: string,
+}
+
+export class NetworkStack extends Stack {
+  public readonly vpc: IVpc;
+
+  public readonly osSecurityGroup: ISecurityGroup;
+
+  constructor(scope: Construct, id: string, props: vpcProps) {
+    let serverAccess: IPeer;
+    super(scope, id, props);
+    if (props.vpcId === undefined) {
+      console.log('No VPC Provided, creating new');
+      this.vpc = new Vpc(this, 'opensearchClusterVpc', {
+        vpcName: 'opensearch-cluster-vpc',
+        cidr: (props.cidrBlock !== undefined) ? props.cidrBlock : '10.0.0.0/16',
+        maxAzs: props.maxAzs,
+        subnetConfiguration: [
+          {
+            name: 'public-subnet',
```

Comment:
`name` here is a mandatory attribute and since it is tied to the VPC, having duplicates will not cause any failure as it is more of a tag than an actual resource name. @gaiksaya